client/lib/wp/sync-handler/index.js

```diff
@@ -121,26 +163,31 @@ export class SyncHandler {
 			 * WP.com server-side response.
 			 *
 			 * @param {Object} serverResponse - server response object
+			 * @returns {Promise} - promise that resolves with server response object
 			 */
 			const cacheResponse = serverResponse => {
-				// get response object without _headers property
-				let responseWithoutHeaders = Object.assign( {}, serverResponse );
-				delete responseWithoutHeaders._headers;
-
-				let storingData = {
-					__sync: {
-						key,
-						synced: Date.now(),
-						syncing: false
-					},
-					body: responseWithoutHeaders,
-					params: reqParams
-				};
-
-				// add/override gotten data from server-side
-				this.storeRecord( key, storingData ).catch( err => {
-					// @TODO error handling
-					warn( err );
+				return new Promise( ( resolve ) => {
+					// get response object without _headers property
+					let responseWithoutHeaders = Object.assign( {}, serverResponse );
+					delete responseWithoutHeaders._headers;
+
+					let storingData = {
+						__sync: {
+							key,
+							synced: Date.now(),
+							syncing: false
+						},
+						body: responseWithoutHeaders,
+						params: reqParams
+					};
+
+					// add/override gotten data from server-side
+					this.storeRecord( key, storingData ).catch( err => {
+						// @TODO error handling
+						warn( err );
+					} );
+
+					resolve( serverResponse );
```

Comment:
@gwwar As I mentioned above, I just wasn't how else to get `checkPagination()` to run inside the `retrieveRecord` promise chain without coupling this functionality with cacheResponse, which I wanted to avoid. I could maybe do this instead:
```js
this.retrieveRecord( key )
  .then( localResponseHandler, recordErrorHandler )
  .then( networkFetch )
  .then( res => {
    cacheResponse( res );
    checkPagination( res );
  }, networkErrorHandler );
```
That'd at least run it at the right time. How would that grab you? or do you have another suggestion?